bazel/foreign_cc/BUILD

```diff
@@ -192,6 +192,109 @@ envoy_cmake_external(
     }),
 )
 
+envoy_cmake_external(
+    name = "llvm",
+    cache_entries = {
+        # Disable both: BUILD and INCLUDE, since some of the INCLUDE
+        # targets build code instead of only generating build files.
+        "LLVM_BUILD_DOCS": "off",
+        "LLVM_INCLUDE_DOCS": "off",
+        "LLVM_BUILD_EXAMPLES": "off",
+        "LLVM_INCLUDE_EXAMPLES": "off",
+        "LLVM_BUILD_RUNTIME": "off",
+        "LLVM_BUILD_RUNTIMES": "off",
+        "LLVM_INCLUDE_RUNTIMES": "off",
+        "LLVM_BUILD_TESTS": "off",
+        "LLVM_INCLUDE_TESTS": "off",
+        "LLVM_BUILD_TOOLS": "off",
+        "LLVM_INCLUDE_TOOLS": "off",
+        "LLVM_BUILD_UTILS": "off",
+        "LLVM_INCLUDE_UTILS": "off",
+        "LLVM_ENABLE_LIBEDIT": "off",
+        "LLVM_ENABLE_LIBXML2": "off",
+        "LLVM_ENABLE_TERMINFO": "off",
+        "LLVM_ENABLE_ZLIB": "off",
+        "LLVM_TARGETS_TO_BUILD": "X86",
+        "CMAKE_CXX_COMPILER_FORCED": "on",
+        # Workaround for the issue with statically linked libstdc++
+        # using -l:libstdc++.a.
+        "CMAKE_CXX_FLAGS": "-lstdc++",
+    },
+    env_vars = {
+        # Workaround for the -DDEBUG flag added in fastbuild on macOS,
+        # which conflicts with DEBUG macro used in LLVM.
+        "CFLAGS": "-UDEBUG",
+        "CXXFLAGS": "-UDEBUG",
+        "ASMFLAGS": "-UDEBUG",
+    },
+    lib_source = "@org_llvm_llvm//:all",
+    static_libraries = select({
```

Comment:
@lizan thoughts on relationship to https://github.com/envoyproxy/envoy/blob/master/tools/clang_tools/support/BUILD.prebuilt?